airflow/ui/src/pages/Variables/AddVariableForm.tsx

```diff
@@ -0,0 +1,131 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Box, Field, HStack, Input, Spacer, Textarea } from "@chakra-ui/react";
+import { useEffect } from "react";
+import { Controller, useForm } from "react-hook-form";
+import { FiSave } from "react-icons/fi";
+
+import { ErrorAlert } from "src/components/ErrorAlert";
+import { Button } from "src/components/ui";
+import { useAddVariable } from "src/queries/useAddVariable";
+
+export type AddVariableBody = {
+  description: string | undefined;
+  key: string;
+  value: string;
+};
+
+type AddVariableFormProps = {
+  readonly onClose: () => void;
+};
+
+const AddVariableForm = ({ onClose }: AddVariableFormProps) => {
+  const { addVariable, error, isPending } = useAddVariable(onClose);
+
+  const {
+    control,
+    formState: { isDirty },
+    handleSubmit,
+    reset,
+    watch,
+  } = useForm<AddVariableBody>({
+    defaultValues: {
+      description: "",
+      key: "",
+      value: "",
+    },
+  });
+
+  const { key, value } = watch();
+
+  useEffect(() => {
+    reset({
+      description: "",
+      key: "",
+      value: "",
+    });
+  }, [reset]);
+
+  const onSubmit = (data: AddVariableBody) => {
+    addVariable(data);
+  };
+
+  return (
+    <>
+      <Controller
+        control={control}
+        name="key"
+        render={({ field }) => (
+          <Field.Root required>
+            <Field.Label fontSize="md">
+              Key <Field.RequiredIndicator />
+            </Field.Label>
+            <Input {...field} required size="sm" />
+          </Field.Root>
+        )}
+      />
+
+      <Controller
+        control={control}
+        name="value"
+        render={({ field }) => (
+          <Field.Root mt={4} required>
+            <Field.Label fontSize="md">
+              Value <Field.RequiredIndicator />
+            </Field.Label>
+            <Textarea {...field} required size="sm" />
```

Comment:
Oh, can you also add the `autoresize` property such that if the user enters more text that the for expands?

See https://www.chakra-ui.com/docs/components/textarea#autoresize
```suggestion
            <Textarea {...field} required size="sm" autoresize />
```

Reply:
Not sure why autoresize is showing me error on this,
`Type '{ autoresize: true; size: "sm"; onChange: (...event: any[]) => void; onBlur: Noop; value: string | undefined; disabled?: boolean | undefined; name: "description"; ref: RefCallBack; }' is not assignable to type 'IntrinsicAttributes & TextareaProps & RefAttributes<HTMLTextAreaElement>'.
  Property 'autoresize' does not exist on type 'IntrinsicAttributes & TextareaProps & RefAttributes<HTMLTextAreaElement>'. Did you mean 'onResize'?ts(2322)`
  
 @bbovenzi do u think we need to add snippet?

Reply:
Strange... not being an React expert... I can confirm same error shows on my side... and it is not working. But is it documented for 3.0.0...

Reply:
I think this can be improved later as well, LGTM!